tests/integration/corrupt-dump-fuzzer.tcl

```diff
@@ -142,10 +143,20 @@ foreach sanitize_dump {no yes} {
 
                 set print_commands false
                 if {!$restore_failed} {
+                    set type [r type "_$k"]
+                    if {$type eq {none}} {
+                        # The key has been removed due to expiration.
+                        # Ensure the process does not crash during expiration and verify
+                        # expire stats to confirm the key was removed due to expiration.
+                        r ping
+                        assert_morethan [s expired_keys] $expired_keys
```

Comment:
i imagined the problem was that the key was deleted during RESTORE (checkAlreadyExpired), but i see HFE doesn't do that, so indeed this is active expire before the TYPE command.

but isn't that stat only affecting active and lazy key expiry, but not HFE?

Reply:
> i imagined the problem was that the key was deleted during RESTORE (checkAlreadyExpired), but i see HFE doesn't do that, so indeed this is active expire before the TYPE command.

no, key was expired in `activeExpireHashFieldCycle()`, not during `RESTORE`.

reproduce steps:
1. add a log in `activeExpireHashFieldCycle()`
2. start server
    `./src/redis-server --enable-debug-command yes`
3. commands:
```
flushdb
config set sanitize-dump-payload yes
debug set-skip-checksum-validation 1
restore key 1 "\x18\xE4\x2D\x8F\xFC\x26\x01\x00\x00\x0A\x01\xC0\x02\xC0\x02\x80\x00\x54\x36\x99\xC0\x06\xC0\x06\x80\x00\x87\x9E\xC1\xC0\x04\xC0\x04\x80\x00\x22\x60\xC9\xC0\x03\x02\x5F\x33\x80\x00\x88\x52\x71\xC0\x05\x02\x5F\x35\x80\x00\x80\x67\x19\xC0\x01\x02\x5F\x31\x80\x00\x0F\xE6\x51\xC0\xB7\xC0\x08\x80\x00\x0D\xF6\x39\xC0\x09\x02\x5F\x39\x80\x00\x54\xA7\xE1\xC0\x00\xC0\x00\x80\x00\x35\xE0\xF9\xC0\x07\x02\x5F\x37\x0C\x00\x7F\x2D\x7A\x68\x26\xA1\x63\x5F" replace
```
4. then we can see log `hash is deleted in hashTypeExpire`

> but isn't that stat only affecting active and lazy key expiry, but not HFE?

yes, should use `stat_expired_subkeys`.